File: backend/app/api/schemas/stats.py
from datetime import date

from pydantic import BaseModel


class DailyStat(BaseModel):
    date: date
    views: int
    vcard_downloads: int


class CardStatsResponse(BaseModel):
    total_views: int
    total_vcard_downloads: int

    views_last_30_days: int
    vcard_downloads_last_30_days: int

    daily: list[DailyStat]

File: frontend/src/types/stats.ts
export interface DailyStat {
  date: string; // ISO формат: "2026-08-01"
  views: number;
  vcard_downloads: number;
}

export interface CardStats {
  total_views: number;
  total_vcard_downloads: number;
  views_last_30_days: number;
  vcard_downloads_last_30_days: number;
  daily: DailyStat[];
}